go-controller/pkg/ovn/address_set/address_set.go

```diff
@@ -30,11 +29,11 @@ type AddressSetFactory interface {
 	// NewAddressSet returns a new object that implements AddressSet
 	// and contains the given IPs, or an error. Internally it creates
 	// an address set for IPv4 and IPv6 each.
-	NewAddressSet(dbIDs *libovsdbops.DbObjectIDs, ips []net.IP) (AddressSet, error)
+	NewAddressSet(dbIDs *libovsdbops.DbObjectIDs, ips sets.Set[string]) (AddressSet, error)
```

Comment:
> Would it make sense to overall change in the names IPs for Addresses?

yes good point because its not IPs anymore

Reply:
> Also it is a bit asymmetric to take a set as input but give a slice as output.
> 
> Chainging the type to a set wouldn't be strictly required or needed by what is described in the commit, de-duplication was already something that was already handled within this utility and probably implies more changes to users?

agreed, hmm de-dup was being done already I just thought if we are changing could go the set route and remove having to do that.

> 
> Also the implementation dances between slices and sets, and is not commited to either so potentially not using the most convenient set specific methods.
> 
> I would probably stick with slices and just change to string type unless you had a reason not to but not evident. You might want to go the other way, I just think we need to commit to one or the other fully.

no I didn't I can do a slice instead of a set since transition from []net.IP to []string{} is easier to digest for users